### ISSUE TITLE
An error when global `this` is undefined

### ISSUE THREAD

**Issue by gchudnov** [Reactive-Extensions/RxJS#850]
Noticed this issue when using Rx with Babel + Browserify and imporing Rx: `import Rx from 'rx'`.

The problem is here:

_/src/core/internal/util.js_:
```javascript
  var inherits = this.inherits = Rx.internals.inherits = function (child, parent) {
    function __() { this.constructor = child; }
    __.prototype = parent.prototype;
    child.prototype = new __();
  };
```
`this.inherits`
^^
`this` might be undefined. It is set-up in _/src/core/headers/outro.js_ as:
```javascript
}.call(this));
```
And top-level `this` is [remapped](https://babeljs.io/docs/faq/#why-is-this-being-remapped-to-undefined-) to `undefined` in Babel.

I fixed it locally by replacing  `}.call(this));` with `}.call(this || {}));`

And now I'm wondering if it is necessary to add `inherits` to the global `this`, polluting it?

**Comment by mattpodwysocki** [comment]
Fixed as per commit and v3.0.1

**Comment by gchudnov** [comment]
Thank you!
